actionpack/test/controller/mime/respond_to_test.rb

```diff
@@ -608,16 +608,20 @@ def test_invalid_format
   end
 
   def test_invalid_variant
+    logger = ActiveSupport::LogSubscriber::TestHelper::MockLogger.new
+    ActionController::Base.logger = logger
+
     @request.variant = :invalid
-    assert_raises(ActionView::MissingTemplate) do
-      get :variant_with_implicit_rendering
-    end
+    get :variant_with_implicit_rendering
+    assert_equal 204, @response.status
```

Comment:
Can we suffice with `assert_response :no_content` here? If so, don't bother with the other assertions in this test.